### DISCUSSION TITLE
Processing Requests Larger than 64k

### DISCUSSION THREAD

**Discussion by skydiablo** [clue/framework-x#234] [Q&A] [answered]
Hi! At the risk of asking the obvious question again, and undoubtedly, there might already be an answer that hasn't helped me so far, I would like to make a new attempt. Why is it (from my perspective) impossible to process a request larger than 64k? I understand why this limit was chosen as the default, but I haven't found a way to adjust this default to suit my needs. Have I overlooked something fundamental, or is there a deliberate effort to prevent this? With somewhat puzzled regards, Volker.

another try from me: https://github.com/clue/framework-x/pull/196

**Comment by SimonFrings** [comment]
Hey @skydiablo, this is a good question which we encountered a few times already, I also recently posted a similar answer in #196. 

We have support for file uploads by loading the file into memory before writing to the filesystem, which works perfectly fine for smaller files, but becomes problematic for larger requests as we should avoid having multiple gigabytes of data stored in memory. To avoid this, ReactPHP provides the option to use streaming request bodies as per https://github.com/reactphp/http#streaming-incoming-request. While this would allow you to accept arbitrary file uploads, it's arguably also somewhat harder to use. With that said, I agree that we should look for a way to make uploading large files easier and somehow create better support for this in ReactPHP.

I know that @clue also took a few attempts at this, so taken from [reactphp/http#428](https://github.com/reactphp/http/issues/428):

> [...] I've looked into this a couple of times already and hope to be able to make this easier in the future, but the main problem that files have to be stored "somewhere" remains for now. I hope to provide some kind of configuration option for this in the future, but no promises at this point [...].

I plan to open a ticket for this in our HTTP component soon and then go further from there on :+1:

**Comment by szado** [comment]
Is there any progress on this? I'd love to use FrameworkX in production, but the lack of support for larger file uploads is holding me back.

**Comment by skydiablo** [comment]
as @SimonFrings mention, just use the streaming feature

**Reply by szado** [reply]
Thanks for the reply. I ended up using it, but I had to fork the entire framework repository just to enable streaming:
https://github.com/szado/framework-x/commit/3184ef59368af6677897818cd095f3c11d8ddf73#diff-68bfd18492ad3475b7967aea2696c0c04e68262c073cc139739872be7d091b38R46
If I didn’t miss anything, it would be great to have this as a built-in feature of the framework 😅

**Comment by clue** [comment, marked as answer]
@skydiablo @szado Just merged #311 which adds experimental support for passing ReactPHP HTTP middleware to the built-in web server, so the 64 KiB limit can now be raised as described in https://framework-x.org/docs/best-practices/deployment/#request-body-limit. This will be part of the upcoming `v0.18.0` release :+1:
